Python 3 updates

zlorb opened the pull request:
Tested with Python 3.8. Some changes were needed to allow work within virtualenv and pyenv.
Test results mostly in-line with reference files.
Contributions by @philscrace2 , @ralic and @zlorb (independently).

ralic committed:
Convert to Python3

philscrace2 committed:
Update the pymodel framework code and samples to use Python3

qgitops committed:
cleanup

qgitops committed:
remove relative imports, fix imp deprecation warning

zlorb committed:
Merge branch 'master' into updates

zlorb committed:
Merge pull request #1 from zlorb/updates

Updates

qgitops committed:
gitignore

zlorb committed:
Update README.md

qgitops committed:
Additional Python 3.x fixes

qgitops committed:
Merge branch 'master' of https://github.com/zlorb/PyModel

qgitops committed:
trun input validation

jon-jacky commented:
This looks impressive!

I haven't worked on PyModel for years and I have never gotten a pull
request for any project.  It's going to take me a while to digest this --
I'll get back to you. -- Jon Jacky


On Sun, Jul 11, 2021 at 2:15 PM Zohar Lorberbaum ***@***.***>
wrote:

> Tested with Python 3.8. Some changes were needed to allow work within
> virtualenv and pyenv.
> Test results mostly in-line with reference files.
> Contributions by @philscrace2 <https://github.com/philscrace2> & @zlorb
> <https://github.com/zlorb> (independently).
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/jon-jacky/PyModel/pull/5
> Commit Summary
>
>    - Convert to Python3
>    - Update the pymodel framework code and samples to use Python3
>    - cleanup
>    - remove relative imports, fix imp deprecation warning
>    - Merge branch 'master' into updates
>    - Merge pull request #1 from zlorb/updates
>    - gitignore
>    - Update README.md
>    - Additional Python 3.x fixes
>    - Merge branch 'master' of https://github.com/zlorb/PyModel
>    - trun input validation
>
> File Changes
>
>    - *A* .gitignore
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (26)
>    - *M* README.md
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (13)
>    - *M* bin/pma
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-9e4e6891d6fcdb3b963f3669bfd36f00c6b0e37d5b730251ce8ec9c8e8556c01>
>    (4)
>    - *M* bin/pmg
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-b74287403e7ba550b6f9b09c34998b0a1f7efed14d9a1ef14e633d326936c508>
>    (4)
>    - *M* bin/pmt
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-c0179db77884e18f4dd6baddc431914410110270a82e9d8317a7b5c7db10a941>
>    (4)
>    - *M* bin/pmv
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-545c4120a17f25731e3daa31dbb30e03a0c38c598ffc448bed790288be993bb5>
>    (4)
>    - *M* bin/trun
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-5777f863f448b357ce290d327ec50831942a0b2b3a9fb59d6ac4c9b5829c5662>
>    (2)
>    - *M* bin/wsgirunner
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-0d7c9e4867594dfd5a61b512c2a0f098e0544d920bb1f44796bc6bd78b9f6c9f>
>    (4)
>    - *M* pymodel/FSM.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-6121b77a550ac28b8525ddfc511afcc4ee8786b03d8d170b7ef252bd5ac6b773>
>    (206)
>    - *M* pymodel/ModelProgram.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-8aaaafd7a0d43bf83f0377716e6c865c534b451ebacc26ab02464c0b152d6e4e>
>    (389)
>    - *M* pymodel/ProductModelProgram.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-a8c3677bf53d31ae11180236f55bd216cea7e2f28b72b000c9c9f467d0ed39dd>
>    (559)
>    - *M* pymodel/TestSuite.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-57679b4c28c368dc09f4cca68b8cd6e479b26b90171a005ecae212d59998c6d6>
>    (235)
>    - *M* pymodel/pma.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-94abd68a19147171d5e10518ab16c993f8f3d7e40ab74f2dedd05674ea51dd34>
>    (10)
>    - *M* pymodel/pmg.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-95c766bc3b50003c7ac3052d0ea5b1c711779baac175f17a715f64962d08b8d5>
>    (4)
>    - *M* pymodel/pmt.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-396950c83604cfc342c666d52762ff85e2028f18bb0f96c4dadfc6200af448d4>
>    (27)
>    - *M* pymodel/pmv.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-1430f77716330f43f40e62f547ab355a9c5d333768eb01fcc25279a9523ff5db>
>    (6)
>    - *M* pymodel/trun.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-8f6ccade3c01f291feaa598ce1fbe72bca53b98d4c64a585848bf1af023161b6>
>    (22)
>    - *M* pymodel/wsgidemo.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-a315131403290dc0277b439b8862884bfb8012590c63f78256a47cc3cd0af45c>
>    (7)
>    - *M* pymodel/wsgirunner.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-002e42e4eaf91e0e25d022eabff8d22602a2105bebe65ee92ff1feb556a6975c>
>    (4)
>    - *M* samples/Socket/stepper.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-83e6bc4a45b0aac8e406866e2a7b2a1019b3fabe674493908278ed8b7a15c0a2>
>    (8)
>    - *M* samples/Socket/stepper_a.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-278e7019686b682d13478f360825f65f9ffc45cff2836bb214c2fb55fd5cf9e0>
>    (8)
>    - *M* samples/Socket/stepper_d.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-5a3b81f1114134750ec374a446f8f149e6801459b1f255e58a26dd7e37e6fce3>
>    (6)
>    - *M* samples/Socket/stepper_o.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-65fa10d68cf5b08dc3c6a5ae60e784cad70a558c268d34fb892265e9d61ec15c>
>    (8)
>    - *M* samples/Socket/stepper_util.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-26af5c0396d0a7a1ae430772b025992a26a09b27da4843ee48a499397e432468>
>    (8)
>    - *M* samples/Timeout/Stepper.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-bddd9e07480a61a517b065f89f93dd428fd9eacebc3bdf9fec7ca380008c343f>
>    (2)
>    - *M* samples/WebApplication/Session.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-7a03a4c612f0a9f71c4b6bd9e09485ce35a6dad6262bb4a546888fa8e52850ec>
>    (160)
>    - *M* samples/WebApplication/Stepper.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-ef0e276b9a5350f8da1d8365d13526bd2ebcfcd04ca507c81a8eb5b7e2a5ee39>
>    (284)
>    - *M* samples/WebApplication/WSGIVerboseConfig.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-f7730e8823b8c24a10a4edaa73e90351e3de3968c9fda9f1266095a80b0f499d>
>    (2)
>    - *M* samples/WebApplication/webapp.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-c4f170c497511c1782ca3638648c0a22d4c452d7dff82b8e8b894aff285ea642>
>    (358)
>    - *M* samples/tracemultiplexer/tracemultiplexer.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-4b43622fb008d690847de8c22ff6ed0229bc7b04ba710ec29ad101c70b5a1303>
>    (410)
>
> Patch Links:
>
>    - https://github.com/jon-jacky/PyModel/pull/5.patch
>    - https://github.com/jon-jacky/PyModel/pull/5.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jon-jacky/PyModel/pull/5>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEKVUW4QLMG7OUHXY22E4DTXICWNANCNFSM5AFU73LQ>
> .
>

jon-jacky commented:
I added a note to the top of my PyModel README linking to your fork.  --
Jon Jacky

On Sun, Jul 11, 2021 at 2:15 PM Zohar Lorberbaum ***@***.***>
wrote:

> Tested with Python 3.8. Some changes were needed to allow work within
> virtualenv and pyenv.
> Test results mostly in-line with reference files.
> Contributions by @philscrace2 <https://github.com/philscrace2> & @zlorb
> <https://github.com/zlorb> (independently).
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/jon-jacky/PyModel/pull/5
> Commit Summary
>
>    - Convert to Python3
>    - Update the pymodel framework code and samples to use Python3
>    - cleanup
>    - remove relative imports, fix imp deprecation warning
>    - Merge branch 'master' into updates
>    - Merge pull request #1 from zlorb/updates
>    - gitignore
>    - Update README.md
>    - Additional Python 3.x fixes
>    - Merge branch 'master' of https://github.com/zlorb/PyModel
>    - trun input validation
>
> File Changes
>
>    - *A* .gitignore
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (26)
>    - *M* README.md
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (13)
>    - *M* bin/pma
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-9e4e6891d6fcdb3b963f3669bfd36f00c6b0e37d5b730251ce8ec9c8e8556c01>
>    (4)
>    - *M* bin/pmg
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-b74287403e7ba550b6f9b09c34998b0a1f7efed14d9a1ef14e633d326936c508>
>    (4)
>    - *M* bin/pmt
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-c0179db77884e18f4dd6baddc431914410110270a82e9d8317a7b5c7db10a941>
>    (4)
>    - *M* bin/pmv
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-545c4120a17f25731e3daa31dbb30e03a0c38c598ffc448bed790288be993bb5>
>    (4)
>    - *M* bin/trun
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-5777f863f448b357ce290d327ec50831942a0b2b3a9fb59d6ac4c9b5829c5662>
>    (2)
>    - *M* bin/wsgirunner
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-0d7c9e4867594dfd5a61b512c2a0f098e0544d920bb1f44796bc6bd78b9f6c9f>
>    (4)
>    - *M* pymodel/FSM.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-6121b77a550ac28b8525ddfc511afcc4ee8786b03d8d170b7ef252bd5ac6b773>
>    (206)
>    - *M* pymodel/ModelProgram.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-8aaaafd7a0d43bf83f0377716e6c865c534b451ebacc26ab02464c0b152d6e4e>
>    (389)
>    - *M* pymodel/ProductModelProgram.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-a8c3677bf53d31ae11180236f55bd216cea7e2f28b72b000c9c9f467d0ed39dd>
>    (559)
>    - *M* pymodel/TestSuite.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-57679b4c28c368dc09f4cca68b8cd6e479b26b90171a005ecae212d59998c6d6>
>    (235)
>    - *M* pymodel/pma.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-94abd68a19147171d5e10518ab16c993f8f3d7e40ab74f2dedd05674ea51dd34>
>    (10)
>    - *M* pymodel/pmg.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-95c766bc3b50003c7ac3052d0ea5b1c711779baac175f17a715f64962d08b8d5>
>    (4)
>    - *M* pymodel/pmt.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-396950c83604cfc342c666d52762ff85e2028f18bb0f96c4dadfc6200af448d4>
>    (27)
>    - *M* pymodel/pmv.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-1430f77716330f43f40e62f547ab355a9c5d333768eb01fcc25279a9523ff5db>
>    (6)
>    - *M* pymodel/trun.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-8f6ccade3c01f291feaa598ce1fbe72bca53b98d4c64a585848bf1af023161b6>
>    (22)
>    - *M* pymodel/wsgidemo.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-a315131403290dc0277b439b8862884bfb8012590c63f78256a47cc3cd0af45c>
>    (7)
>    - *M* pymodel/wsgirunner.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-002e42e4eaf91e0e25d022eabff8d22602a2105bebe65ee92ff1feb556a6975c>
>    (4)
>    - *M* samples/Socket/stepper.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-83e6bc4a45b0aac8e406866e2a7b2a1019b3fabe674493908278ed8b7a15c0a2>
>    (8)
>    - *M* samples/Socket/stepper_a.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-278e7019686b682d13478f360825f65f9ffc45cff2836bb214c2fb55fd5cf9e0>
>    (8)
>    - *M* samples/Socket/stepper_d.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-5a3b81f1114134750ec374a446f8f149e6801459b1f255e58a26dd7e37e6fce3>
>    (6)
>    - *M* samples/Socket/stepper_o.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-65fa10d68cf5b08dc3c6a5ae60e784cad70a558c268d34fb892265e9d61ec15c>
>    (8)
>    - *M* samples/Socket/stepper_util.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-26af5c0396d0a7a1ae430772b025992a26a09b27da4843ee48a499397e432468>
>    (8)
>    - *M* samples/Timeout/Stepper.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-bddd9e07480a61a517b065f89f93dd428fd9eacebc3bdf9fec7ca380008c343f>
>    (2)
>    - *M* samples/WebApplication/Session.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-7a03a4c612f0a9f71c4b6bd9e09485ce35a6dad6262bb4a546888fa8e52850ec>
>    (160)
>    - *M* samples/WebApplication/Stepper.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-ef0e276b9a5350f8da1d8365d13526bd2ebcfcd04ca507c81a8eb5b7e2a5ee39>
>    (284)
>    - *M* samples/WebApplication/WSGIVerboseConfig.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-f7730e8823b8c24a10a4edaa73e90351e3de3968c9fda9f1266095a80b0f499d>
>    (2)
>    - *M* samples/WebApplication/webapp.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-c4f170c497511c1782ca3638648c0a22d4c452d7dff82b8e8b894aff285ea642>
>    (358)
>    - *M* samples/tracemultiplexer/tracemultiplexer.py
>    <https://github.com/jon-jacky/PyModel/pull/5/files#diff-4b43622fb008d690847de8c22ff6ed0229bc7b04ba710ec29ad101c70b5a1303>
>    (410)
>
> Patch Links:
>
>    - https://github.com/jon-jacky/PyModel/pull/5.patch
>    - https://github.com/jon-jacky/PyModel/pull/5.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jon-jacky/PyModel/pull/5>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEKVUW4QLMG7OUHXY22E4DTXICWNANCNFSM5AFU73LQ>
> .
>

zlorb committed:
Merge branch 'master' into master

Commit:
The set type is no longer ordered in Python 3.7+, meaning that pmt actions
are no longer reproducible with non-zero seeds.  Use fromkeys and list as
a workaround (another solution would be ordered-set from PyPi).

Note there are multiple other occurrences of sets remaining, which may
cause ordering issues elsewhere - the same workaround could be applied to
them if necessary.

zlorb committed:
Merge pull request #2 from jproudlo/setReplacement

The set type is no longer ordered in Python 3.7+, meaning that pmt ac…

zlorb committed:
Update README.md

zlorb committed:
Update README.md
